New source - Shipstation

kevjunior1990 opened the pull request:
## What
Providing a new source connector for users that want to make use of the Shipstation API list within their projects

## How
The source was implemented using the Airbyte Low Code generator

## Review guide
<!--
1. `x.py`
2. `y.py`
-->

## User Impact
The user will be able to call on the following list of API's from Shipstation:

    - "account_tags"
    - "carriers"
    - "customers"
    - "fulfillments"
    - "orders"
    - "packages"
    - "products"
    - "services"
    - "shipments"
    - "stores"
    - "users"
    - "warehouses"
    - "webhooks"

## Can this PR be safely reverted and rolled back?
- [ ] YES 💚

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/airbytehq/airbyte?pullRequest=46568) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/airbytehq/airbyte?pullRequest=46568) before we can accept your contribution.<br/><hr/>**Kevin O Sullivan** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/airbytehq/airbyte?pullRequest=46568) it.</sub>

natikgadzhi commented:
Thank you for this! 

Bad news: this should be a manifest-only connector that you can make in the Connector Builder in Airbyte Cloud or OSS instead. 35 files is way too much boilerplate for this.

kevjunior1990 commented:
Thank you for the feedback on the Shipstation source PR.

With regards to your comment that there are too many files associated with this source connector, I don't think that I have introduced boilerplate code. The majority of the additional files that I have added are to read a json schema associated with each stream from individual files instead of reading from an inline schema in the manifest.yaml file

For me, decoupling the schema from the manifest and reading the schema from a json file associated with an individual stream is a more maintainable way of adding a source rather than packaging all the definitions into a single manifest file.

Please let me know if my thought process makes sense and I would be happy to discuss further if needed

marcosmarxm commented:
@kevjunior1990 at the moment the project is only accepting new connectors made in the connector builder to make maintenance and improvements easy and fast. You're welcome to resubmit your contribution building the connector direclty in the UI and publish from there.
